templates/default/sqlserver.yaml.erb

```diff
@@ -1,39 +1,9 @@
-init_config:
-    #
-    # By default, we only capture *some* of the metrics available in the
-    # `sys.dm_os_performance_counters` table. You can easily add additional
-    # metrics by following the custom_metrics structure shown below.
-    #
-    #custom_metrics:
-    #
-    #    This is a basic custom metric. There is not instance assicated with
-    #    this counter.
-    #    Options for type are: gauge, rate, histogram
-    #
-    #    -   name: sqlserver.clr.execution
-    #        type: gauge
-    #        counter_name: CLR Execution
-    #
-    #    This counter has multiple instances assocaited with it and we're
-    #    choosing to only fetch the '_Total' instance.
-    #
-    #    -   name: sqlserver.exec.in_progress
-    #        type: gauge
-    #        counter_name: OLEDB calls
-    #        instance_name: Cumulative execution time (ms) per second
-    #
-    #    This counter has multiple instances assocaited with it and we want
-    #    every instance available. We'll use the special case ALL instance
-    #    which *requires* a value for "tag_by". In this case, we'll get metrics
-    #    tagged as "db:mydb1", "db:mydb2".
-    #
-    #    -   name: sqlserver.db.commit_table_entries
-    #        type: gauge
-    #        counter_name: Log Flushes/sec
-    #        instance_name: ALL
-    #        tag_by: db
+# Generated by Chef, local modifications will be overwritten
 
-instances:
-    -   host: HOST,PORT
-        username: my_username
-        password: my_password
+<% if @init_config.empty? -%>
```

Comment:
I did try that, however when I just use `<%= JSON.parse(({'init_config' => @init_config, 'instances' => @instances}).to_json).to_yaml %>`, when `@init_config` is empty, the chef spec test fails with:
```
1) datadog::sqlserver config 2 renders expected YAML config file
     Failure/Error: expect(YAML.load(content).to_json).to be_json_eql(YAML.load(expected_yaml).to_json)
     
       Expected equivalent JSON
       Diff:
       @@ -1,5 +1,6 @@
        {
       -  "init_config": null,
       +  "init_config": {
       +  },
          "instances": [
            {
              "command_timeout": 30,
       
     # ./spec/integrations/sqlserver_spec.rb:135:in `block (4 levels) in <top (required)>'
     # /home/coopem67/.chefdk/gem/ruby/2.3.0/gems/chefspec-5.4.0/lib/chefspec/matchers/render_file_matcher.rb:120:in `matches_content?'
     # /home/coopem67/.chefdk/gem/ruby/2.3.0/gems/chefspec-5.4.0/lib/chefspec/matchers/render_file_matcher.rb:13:in `matches?'
     # ./spec/integrations/sqlserver_spec.rb:134:in `block (3 levels) in <top (required)>'

Finished in 5.16 seconds (files took 5.76 seconds to load)
20 examples, 1 failure

Failed examples:

rspec ./spec/integrations/sqlserver_spec.rb:133 # datadog::sqlserver config 2 renders expected YAML config file
```